pkg/operator/ceph/cluster/controller.go

```diff
@@ -271,27 +271,28 @@ func (c *ClusterController) onAdd(obj interface{}) {
 		logger.Warningf("mon count is even (given: %d), should be uneven, continuing", cluster.Spec.Mon.Count)
 	}
 
-	cephVersion, err := cluster.detectCephVersion(cluster.Spec.CephVersion.Image, 15*time.Minute)
-	if err != nil {
-		logger.Errorf("unknown ceph major version. %+v", err)
-		return
-	}
+	// Start the Rook cluster components. Retry several times in case of failure.
+	err = wait.Poll(clusterCreateInterval, clusterCreateTimeout, func() (bool, error) {
+		cephVersion, err := cluster.detectCephVersion(cluster.Spec.CephVersion.Image, 15*time.Minute)
+		if err != nil {
+			logger.Errorf("unknown ceph major version. %+v", err)
+			return false, nil
+		}
 
-	if !cluster.Spec.CephVersion.AllowUnsupported {
-		if !cephVersion.Supported() {
-			logger.Errorf("unsupported ceph version detected: %s. allowUnsupported must be set to true to run with this version.", cephVersion)
-			return
+		if !cluster.Spec.CephVersion.AllowUnsupported {
+			if !cephVersion.Supported() {
+				logger.Errorf("unsupported ceph version detected: %s. allowUnsupported must be set to true to run with this version.", cephVersion)
+				// it might sound strange to log error and exit true but we don't want to retry if the version is not supported
```

Comment:
This looks correct to return true. However, I think we'll need to set a separate flag that indicates there was actually failure. For example, if we set a new var `validOrchestration = false`, then around line 309 we would return in that case instead of continuing.